src/dodal/devices/zebra.py

```diff
@@ -45,99 +33,125 @@
 TTL_PANDA = 4
 
 
-class I03Axes(Enum):
-    SMARGON_X1 = "Enc1"
-    SMARGON_Y = "Enc2"
-    SMARGON_Z = "Enc3"
-    OMEGA = "Enc4"
+class ArmSource(str, Enum):
+    SOFT = "Soft"
+    EXTERNAL = "External"
+
+
+class TrigSource(str, Enum):
+    POSITION = "Position"
+    TIME = "Time"
+    EXTERNAL = "External"
+
+
+class EncEnum(str, Enum):
+    Enc1 = "Enc1"
+    Enc2 = "Enc2"
+    Enc3 = "Enc3"
+    Enc4 = "Enc4"
+    Enc1_4Av = "Enc1-4Av"
 
 
-class I24Axes(Enum):
-    VGON_Z = "Enc1"
-    OMEGA = "Enc2"
-    VGON_X = "Enc3"
-    VGON_YH = "Enc4"
+class I03Axes:
+    SMARGON_X1 = EncEnum.Enc1
+    SMARGON_Y = EncEnum.Enc2
+    SMARGON_Z = EncEnum.Enc3
+    OMEGA = EncEnum.Enc4
 
 
-class RotationDirection(IntEnum):
-    POSITIVE = 1
-    NEGATIVE = -1
+class I24Axes:
+    VGON_Z = EncEnum.Enc1
+    OMEGA = EncEnum.Enc2
+    VGON_X = EncEnum.Enc3
+    VGON_YH = EncEnum.Enc4
 
 
-class ArmDemand(IntEnum):
+class RotationDirection(str, Enum):
+    POSITIVE = "Positive"
+    NEGATIVE = "Negative"
+
+
+class ArmDemand(Enum):
     ARM = 1
     DISARM = 0
 
 
-class FastShutterAction(IntEnum):
-    OPEN = 1
-    CLOSE = 0
+class SoftInState(str, Enum):
+    YES = "Yes"
+    NO = "No"
 
 
-class ArmingDevice(Device):
+class ArmingDevice(StandardReadable):
     """A useful device that can abstract some of the logic of arming.
     Allows a user to just call arm.set(ArmDemand.ARM)"""
 
     TIMEOUT = 3
 
-    arm_set = Component(EpicsSignal, "PC_ARM")
-    disarm_set = Component(EpicsSignal, "PC_DISARM")
-    armed = Component(EpicsSignal, "PC_ARM_OUT")
+    def __init__(self, prefix: str, name: str = "") -> None:
+        self.arm_set = epics_signal_rw(float, prefix + "PC_ARM")
+        self.disarm_set = epics_signal_rw(float, prefix + "PC_DISARM")
+        self.armed = epics_signal_rw(float, prefix + "PC_ARM_OUT")
+        super().__init__(name)
 
-    def set(self, demand: ArmDemand) -> StatusBase:
-        status = await_value(self.armed, demand.value, timeout=self.TIMEOUT)
+    async def _set_armed(self, demand: ArmDemand):
+        await self.armed.set(demand.value)
```

Comment:
Should: I think `armed` is a read-only signal from EPICS as to whether the zebra thinks it's armed. We can't set it. Instead I think we want to observe it after we've done the set and wait for it to be as we expect. See other places we use `observe_value`